server/controllers/users.js

```diff
@@ -0,0 +1,78 @@
+import jwt from "jsonwebtoken";
+import users from "../db/users";
+class UsersController {
+  getAllUsers(req, res) {
+    res.status(200).json({
+      status: 200,
+      data: users
+    });
+  }
+
+  signup(req, res) {
+    const {
+      email,
+      firstname,
+      lastname,
+      password,
+      address,
+      status,
+      isAdmin
+    } = req.body;
+
+    const userExists = users.find(user => user.email === email);
+
+    if (userExists) {
+      res.status(400).json({
+        status: 400,
+        error: "User already exists"
+      });
+    }
+
+    const user = {};
+    user.id = users.length + 1;
+    user.email = email;
+    user.firstname = firstname;
+    user.lastname = lastname;
+    user.password = password;
+    user.address = address;
+    user.status = "pending" || "approved";
+    user.isAdmin = true || false;
+    users.push(user);
+    jwt.sign(
+      {
+        userId: users.id,
+        email: users.email,
+        password: users.password
+      },
+      "secretkey",
+      (error, token) => {
+        if (error) {
+          return res.status(401).json({
```

Comment:
The status code here is wrong. Did you read the document I shared on the channel about status codes?

Reply:
Yes, I read them and understood very well.